agent/xds/clusters.go

```diff
@@ -1141,6 +1144,23 @@ func (s *ResourceGenerator) makeUpstreamClustersForDiscoveryChain(
 				OutlierDetection: ToOutlierDetection(upstreamConfig.PassiveHealthCheck),
 			}
 
+			// Lookup listener and service config details from ingress gateway
+			// definition.
+			var svc *structs.IngressService
+			if lCfg, ok := cfgSnap.IngressGateway.Listeners[listenerKey]; ok {
+				svc = findIngressServiceMatchingUpstream(lCfg, *upstream)
+			}
+
+			if svc != nil && svc.MaxConnections > 0 {
+				c.CircuitBreakers = &envoy_cluster_v3.CircuitBreakers{
+					Thresholds: []*envoy_cluster_v3.CircuitBreakers_Thresholds{
+						{
+							MaxConnections: makeUint32Value(int(svc.MaxConnections)),
+						},
+					},
+				}
+			}
```

Comment:
How will this interact with the Thresholds being set in L1142? Do we simply overwrite values from `upstreamConfig.Limits`?